x-pack/plugins/ml/public/application/explorer/alerts/alerts_panel.tsx

```diff
@@ -0,0 +1,96 @@
+import { AlertConsumers } from '@kbn/rule-data-utils';
+import React, { useState, type FC } from 'react';
+import useObservable from 'react-use/lib/useObservable';
+import { ML_ALERTS_CONFIG_ID } from '../../../alerting/anomaly_detection_alerts_table/register_alerts_table_configuration';
+import { CollapsiblePanel } from '../../components/collapsible_panel';
+import { useMlKibana } from '../../contexts/kibana';
+import { useAnomalyExplorerContext } from '../anomaly_explorer_context';
+import { AlertsSummary } from './alerts_summary';
+import { AnomalyDetectionAlertsOverviewChart } from './chart';
+import { statusNameMap } from './const';
+
+export const AlertsPanel: FC = () => {
+  const {
+    services: { triggersActionsUi },
+  } = useMlKibana();
+
+  const [isOpen, setIsOpen] = useState(true);
+  const [toggleSelected, setToggleSelected] = useState(`alertsSummary`);
+
+  const { anomalyDetectionAlertsStateService } = useAnomalyExplorerContext();
+
+  const countByStatus = useObservable(anomalyDetectionAlertsStateService.countByStatus$);
+  const alertsQuery = useObservable(anomalyDetectionAlertsStateService.alertsQuery$, {});
+
+  const alertStateProps = {
+    alertsTableConfigurationRegistry: triggersActionsUi!.alertsTableConfigurationRegistry,
+    configurationId: ML_ALERTS_CONFIG_ID,
+    id: `ml-details-alerts`,
+    featureIds: [AlertConsumers.ML],
+    query: alertsQuery,
+    showExpandToDetails: true,
+    showAlertStatusWithFlapping: true,
+  };
+  const alertsStateTable = triggersActionsUi!.getAlertsStateTable(alertStateProps);
+
+  const toggleButtons = [
+    {
+      id: `alertsSummary`,
+      label: i18n.translate('xpack.ml.explorer.alertsPanel.summaryLabel', {
+        defaultMessage: 'Summary',
+      }),
+    },
+    {
+      id: `alertsTable`,
+      label: 'Details',
+    },
+  ];
+
+  return (
+    <>
+      <CollapsiblePanel
+        isOpen={isOpen}
+        onToggle={setIsOpen}
+        header={
+          <FormattedMessage id="xpack.ml.explorer.alertsPanel.header" defaultMessage="Alerts" />
+        }
+        headerItems={Object.entries(countByStatus ?? {}).map(([status, count]) => {
+          return (
+            <>
+              {statusNameMap[status]}{' '}
+              <EuiNotificationBadge size="m" color={status === 'active' ? 'accent' : 'subdued'}>
+                {count}
+              </EuiNotificationBadge>
+            </>
+          );
+        })}
+      >
+        <AnomalyDetectionAlertsOverviewChart />
+
+        <EuiSpacer size="m" />
+        <EuiButtonGroup
+          legend={i18n.translate('xpack.ml.explorer.alertsPanel.summaryTableToggle', {
+            defaultMessage: 'Summary / Table view toggle',
+          })}
+          options={toggleButtons}
+          idSelected={toggleSelected}
+          onChange={setToggleSelected}
+        />
+        <EuiSpacer size="m" />
+
+        {toggleSelected === 'alertsTable' ? alertsStateTable : <AlertsSummary />}
```

Comment:
Do you know why 'Add to new case' and 'Add to existing case' aren't appearing in the bulk actions menu?

<img width="263" alt="image" src="https://github.com/elastic/kibana/assets/7405507/39a8daa7-03f4-4917-a2e6-3af498f2bce1">

Reply:
I assume this action isn't supported as bulk, cc @XavierM

Reply:
you need to do some manual work to get it working. @umbopepato and I are refactoring this to make it easier. To avoid circular dependency between triggerActionsUI and Cases, we need to do some refactor to remove the cases action in a packages so we can avoid the circular dependency between plugins.

Reply:
That would be awesome. Ideally, we should add the case action in the row action automatically as we do with the bulk actions.